test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,297 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const { getContractAt } = ethers;
+const { expect } = require('chai');
+const { getRandomBytes32 } = require('./utils');
+const { deployContract } = require('../scripts/deploy');
+
+/**
+ * Helper function to extract an address from a bytes32 storage slot
+ * @param {string} slotValue - The bytes32 value from storage
+ * @returns {string} The EIP-55 checksummed address
+ */
+function bytes32ToAddress(slotValue) {
```

Comment:
```suggestion
function bytes32ToAddress(bytes) {
```
the helper doesn't care where the value comes from, only that it's a bytes32 value